bitburner-official/bitburner-src
# BUGFIX: Running scripts may be loaded before main UI

When we load running scripts, those scripts may run before the main UI is loaded. In that case, anything that interacts with the router will fail to run. For example, Singularity APIs that call `Router.toPage()` will throw an error, and that error may not be shown to the player because the main UI is not loaded.

How to reproduce:
- Join Daedalus.
- Run test code.
- Reload and check the console.

If you cannot reproduce it consistently, you can forcefully delay the main UI: In `src\ui\LoadingScreen.tsx`, add another promise (e.g., `new Promise((resolve) => setTimeout(resolve, 1000))`) to `[initSwc(), Engine.load(saveData)]`.

Test code:
```js
/** @param {NS} ns */
export async function main(ns) {
  try {
    ns.singularity.workForFaction("Daedalus", "hacking");
  } catch (e) {
    console.error(e);
  }
  while (true) {
    await ns.sleep(1000);
  }
}
```
Closes #1707.
